spec/support/energy_tariffs_shared_examples.rb

```diff
@@ -23,6 +23,19 @@
   it 'navigates the index tabs' do
     expect(current_path).to end_with('energy_tariffs')
 
+    if tariff_holder.school?
```

Comment:
This test is trying to do too may things... can it be split in to different tests / or contexts per tariff holder type? It's a big one to refactor I know but it's quite hard to read as is.

Reply:
agreed.  this is one of the consequences of using shared examples everywhere (another being not able to look at the tests and easily follow what the tests are actually doing and easily isolate where they fail when they do).  

there's a whole bunch of cases in `spec/support/energy_tariffs_shared_examples.rb` where there are behaviours conditional to tariff holder or current user and really tricky to pick apart (and this is also why i often prefer duplication of code over wrong abstractions). 

I think here, since we're testing correct behaviour of navigating the index tabs, we might have to let this one go and address a big test refactor in a different pr.  sound ok?